### ISSUE TITLE
wasmate generates incorrect call_indirect

### ISSUE THREAD

**Issue by binji** [WebAssembly/experimental#62]
See 20021118-2.c.s in LLVM torture tests:

```
	f32.const	$push7=, 0x1.4p1
	f32.const	$push6=, 0x1.cp1
	call_indirect	$3, $pop7, $pop6
```
is translated to
```
    (call_indirect (get_local 3) (f32.const 0x1.4p1) (f32.const 0x1.cp1))
```

But s-expression format requires function type signature to be specified. No function table is generated either.

**Comment by ncbray** [comment]
In general, function pointers are borked.  I was looking into specifying a function table in sexprs, but there doesn't appear to be an existing precedent.  The "linker" needs to generate the function table so it can embed consistent function pointer values in the data segment, as constants, etc.

**Comment by jfbastien** [comment]
I added support for function table generation in #75.

I believe that I generate the same thing as [this spec test](https://github.com/WebAssembly/spec/blob/master/ml-proto/test/func_ptrs.wast) for the table, but #75 only addressed address-taken functions being relocated to the `.data` section. This was triggered by a few torture tests which stored function pointers in globals.

I just added [an LLVM test with vtables](http://reviews.llvm.org/rL256102) which should allow us to test how vtables work.

I haven't tried out `call_indirect` yet, so I'm not marking this issue as fixed.
